Automattic/jetpack
# Search: Remove unexpected sidebar widgets added on theme switch in WPCOM

Fixes #18556

#### Changes proposed in this Pull Request:
#19873 fixed the 'unwanted widgets' issue when switching theme for wp.org sites, but for wpcom the issue still exists. The reason is #19873 fixed the issue only when user trigger theme switch in admin panel, while wpcom triggers theme switch by API.

This PR moves the action to run at both front end and admin panel, and as it'll potentially affect all users while switching themes, the condition to run the fix is further restricted.

#### Jetpack product discussion
pcNPJE-96-p2

#### Does this pull request change what data or activity we track or use?
no

#### Testing instructions:
- Apply this patch D61881 to sandbox and set up hosts to sandbox public-api
- Go to a wpcom *simple* site with Jetpack Search subscription
- Switch theme to Twenty Twenty which has two sidebar area ("Footer #1" and "Footer #2"), make sure both sidebars have some widgets
- Switch to a theme that has less sidebars, say Twenty Twenty-One which has only one sidebar ("Footer").
- Ensure that the Jetpack Search sidebar has the same widgets before the theme switch, meaning it doesn't have a bunch of unwanted widgets (set in step 3), like 'Archives', 'Categories' and 'Meta' etc.
